### DISCUSSION TITLE
Speech recognition tool

### DISCUSSION THREAD

**Discussion by alfred-SL** [w3c/wcag#4136] [Q&A] [answered]
Is speech recognition tool (e.g. Dragon) considered while evaluating WCAG criteria? Is it part of WCAG? What are the considerations during testing?

**Comment by TestPartners** [comment, marked as answer]
WCAG SC 2.5.3 (Label In Name) was created for the benefit of speech recognition software users, but there is no requirement to use that software when assessing conformance. In fact, you definitely should not do so because Dragon uses heuristics that will give the wrong results (it can give a pass instead of a non-conformance, but not the opposite).

That success criterion only addresses one of the barriers that speech recognition software users can encounter, and the other barriers are not currently addressed by any success criteria.

Whether you do any testing with Dragon or other products depends on the objective of your testing. If you are only testing for WCAG conformance, there is no need to test with those products. If you want to make the software as accessible as possible, then you should test with them.

If you don’t have experience of user testing with Dragon users, you can’t be sure what will and will not be a problem for them. In fact, this applies to all assistive technologies. I recommend that you start by running or observing some user testing sessions. If you don’t have a real project to do this on, do it as a research project, which is what we did 20 years ago. It’s a lot easier now that the testing can be done remotely, and Dragon isn’t as flaky as it used to be.

Steve Green
Managing Director
Test Partners Ltd

From: alfred-SL ***@***.***>
Sent: 06 November 2024 14:53
To: w3c/wcag ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [w3c/wcag] Speech recognition tool (Discussion #4136)


Is speech recognition tool (e.g. Dragon) considered while evaluating WCAG criteria? Is it part of WCAG? What are the considerations during testing?

**Reply by patrickhlauke** [reply]
> That success criterion only addresses one of the barriers that speech recognition software users can encounter, and the other barriers are not currently addressed by any success criteria.

[2.1.4 Character Key Shortcuts](https://www.w3.org/WAI/WCAG22/Understanding/character-key-shortcuts.html) also tries to (on its face at least) address a speech user problem (of accidentally triggering functionality in the page)
